tools/clangTidyCheck.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+if type clang-tidy > /dev/null 2>&1; then
+	CLANGTIDY="clang-tidy";
+elif type clang-tidy > /dev/null 2>&1; then
+	CLANGTIDY="clang-tidy-9";
+else
+	echo "clang-tidy not installed!";
+	exit 1;
+fi
+
+# Update compile_commands.json file
+mkdir -p build
+cd build
+cmake .. -DCMAKE_EXPORT_COMPILE_COMMANDS=1
+mv compile_commands.json ..
+cd ..
```

Comment:
Running `cmake ..` requires that Qt is installed in workflow, if you want to skip this step. We can put actual compile_commands.json inside our source root. We can use linux CI to check for up to date `compile_commands.json`, just like we check for up to date translations.

A drawback would be that clang-tidy checks could only be performed in linux so any block of code with OS specifier like windows, macOS will not be performed.